src/util/events.ts

```diff
@@ -16,95 +17,72 @@ import { ScrollView } from './scroll-view';
  * constructor(public events: Events) {}
  * createUser(user) {
  *   console.log('User created!')
- *   events.publish('user:created', user, Date.now());
+ *   events.publish('user:created', { user, time: Date.now() });
  * }
  *
  *
  * // second page (listen for the user created event after function is called)
  * constructor(public events: Events) {
- *   events.subscribe('user:created', (user, time) => {
- *     // user and time are the same arguments passed in `events.publish(user, time)`
+ *   events.subscribe('user:created', ({ user, time }) => {
+ *     // user and time are the same arguments passed as keys in `events.publish({ ... })`
  *     console.log('Welcome', user, 'at', time);
  *   });
  * }
  *
+ * // third page utilises rxjs interface
+ * constructor(public events: Events) {
+ *   this.destroyed = new Subject<boolean>();
+ *   events.topic('user:created')
+ *     .takeUntil(this.destroyed)
+ *     .subscribe(({ user, time }) => console.log('Welcome', user, 'at', time));
+ * }
+ *
  * ```
  * @demo /docs/demos/src/events/
  */
 export class Events {
-  private _channels: any = [];
+  private _channels: { [topic: string]: Subject<any> } = {};
```

Comment:
Frankly speaking, don't see the purpose. Typing is almost the same, also it's internal state representation. Could you please clarify why you think it would better to change object to `Map`.